src/Appium.Net/Appium/Service/AppiumServiceBuilder.cs

```diff
@@ -306,6 +307,33 @@ public AppiumServiceBuilder WithStartUpTimeOut(TimeSpan startUpTimeout)
             return this;
         }
 
+        /// <summary>
+        /// Command line arguments that will be passed to NodeJS when it starts up.
+        /// </summary>
+        /// <param name="arguments">
+        /// A collection of argument name and argument value pairs that will be
+        /// passed to NodeJS.
+        /// </param>
+        /// <returns></returns>
+        public AppiumServiceBuilder WithNodeArguments(IDictionary<string, string> arguments)
```

Comment:
arguments are not necessarily pairs, there also might be single switchers

Reply:
This is the same pattern used in `OptionsCollector.AddArguments` and `AppiumServiceBuilder.WithEnvironment`. If you prefer a list I can change it. Not much difference either way.

Reply:
for env dictionary makes sense, because there all entries are naturally key-value pairs, which is not the case for command line options

Reply:
Changed in latest